src/api.cpp

```diff
@@ -2111,6 +2137,21 @@ cl_int CLVK_API_CALL clGetMemObjectInfo(cl_mem mem, cl_mem_info param_name,
         copy_ptr = memobj->properties().data();
         ret_size = memobj->properties().size() * sizeof(cl_mem_properties);
         break;
+    case CL_MEM_DEVICE_PTR_EXT: {
+        auto buffer = static_cast<cvk_buffer*>(memobj);
+        if (!buffer->is_buffer_type()) {
+            ret = CL_INVALID_MEM_OBJECT;
+            break;
+        }
+        if (!buffer->context()->device()->supports_buffer_device_address()) {
+            ret = CL_INVALID_OPERATION;
+            break;
+        }
+        val_sizet = buffer->device_address();
```

Comment:
The spec defines a new type that aliases `cl_ulong` which, contrary to `size_t` is guaranteed to have a size of 64 bits.